static/app/views/automations/components/automationBuilder.tsx

```diff
@@ -0,0 +1,250 @@
+    groupIndex: number,
+    conditionIndex: number,
+    comparison: any
+  ) => void;
+  updateIfLogicType: (groupIndex: number, logicType: DataConditionGroupLogicType) => void;
+}
+
+function ActionFilterBlock({
+  id,
+  onDelete,
+  addIfCondition,
+  removeIfCondition,
+  updateIfCondition,
+  updateIfLogicType,
+}: ActionFilterBlockProps) {
+  const {state} = useAutomationBuilderContext();
+  const actionFilterBlock = state.actionFilters[id];
+
+  return (
+    <IfThenWrapper key={id}>
+      <Step>
+        <Flex column gap={space(0.75)}>
+          <Flex justify="space-between">
+            <StepLead>
+              {tct('[if: If] [selector] of these filters match', {
+                if: <Badge />,
+                selector: (
+                  <EmbeddedWrapper>
+                    <EmbeddedSelectField
+                      styles={{
+                        control: (provided: any) => ({
+                          ...provided,
+                          minHeight: '21px',
+                          height: '21px',
+                        }),
+                      }}
+                      inline={false}
+                      isSearchable={false}
+                      isClearable={false}
+                      name={`actionFilters.${id}.logicType`}
+                      required
+                      flexibleControlStateSize
+                      options={FILTER_MATCH_OPTIONS}
+                      size="xs"
+                      value={actionFilterBlock?.logicType}
+                      onChange={value => updateIfLogicType(id, value)}
+                    />
+                  </EmbeddedWrapper>
+                ),
+              })}
+            </StepLead>
+            <DeleteButton
+              aria-label={t('Delete If/Then Block')}
+              size="sm"
+              icon={<IconDelete />}
+              borderless
+              onClick={onDelete}
+            />
+          </Flex>
+          <RuleNodeList
+            placeholder={t('Filter by...')}
+            group={`actionFilters.${id}`}
+            conditions={actionFilterBlock?.conditions || []}
+            onAddRow={type => addIfCondition(id, type)}
+            onDeleteRow={index => removeIfCondition(id, index)}
+            updateCondition={(index, comparison) =>
+              updateIfCondition(id, index, comparison)
+            }
+          />
+        </Flex>
+      </Step>
+      <Step>
+        <StepLead>
+          {tct('[then:Then] perform these actions', {
+            then: <Badge />,
+          })}
+        </StepLead>
+        {/* TODO: add actions dropdown here */}
+      </Step>
+    </IfThenWrapper>
+  );
+}
+
+const PurpleTextButton = styled(Button)`
+  color: ${p => p.theme.purple300};
+  font-weight: normal;
+  padding: 0;
+`;
+
+const Step = styled(Flex)`
+  flex-direction: column;
+  gap: ${space(0.75)};
+`;
+
+const StepLead = styled(Flex)`
+  align-items: center;
+  gap: ${space(0.5)};
+`;
+
+const Badge = styled('span')`
```

Comment:
Can we use `<Tag type="highlight" />` instead of this custom component? I'll double check with design.

Reply:
hmm they look pretty different, would depend on what design says!

<img width="212" alt="Screenshot 2025-04-16 at 1 59 48 PM" src="https://github.com/user-attachments/assets/87512871-ba53-43a7-bee3-f9001c61551a" />
<img width="595" alt="Screenshot 2025-04-16 at 2 00 13 PM" src="https://github.com/user-attachments/assets/0a374df6-884d-41ea-983f-d7f8cd4ec4fb" />

Reply:
We can punt on this, not a big deal.